ci/jenkins/test.sh

```diff
@@ -138,31 +138,59 @@ function clean_for_windows_install_cni {
-        ssh -o StrictHostKeyChecking=no -n Administrator@${IP} "ipconfig.exe" > "network_info/${IP}/ipconfig" || true
-        ssh -o StrictHostKeyChecking=no -n Administrator@${IP} "powershell.exe Get-HNSNetwork" > "network_info/${IP}/hns_network" || true
-        ssh -o StrictHostKeyChecking=no -n Administrator@${IP} "powershell.exe Get-HNSEndpoint" > "network_info/${IP}/hns_endpoint" || true
+        IP=$(kubectl get pod ${AGENTNAME} -n kube-system -o json | jq -r '.status.hostIP')
+        mkdir "${DEBUG_LOG_PATH}/${AGENTNAME}"
+
+        echo "=== Collecting '${AGENTNAME}' agent network info after failure ==="
+        AGENT_NETWORK_INFO_PATH="${DEBUG_LOG_PATH}/${AGENTNAME}/network_info"
+        mkdir "${AGENT_NETWORK_INFO_PATH}"
+        if [[ "${AGENTNAME}" =~ "windows" ]]; then
+            ssh -o StrictHostKeyChecking=no -n administrator@${IP} "ovs-ofctl dump-flows br-int" > "${AGENT_NETWORK_INFO_PATH}/flows" || true
+            ssh -o StrictHostKeyChecking=no -n administrator@${IP} "ovs-vsctl show" > "${AGENT_NETWORK_INFO_PATH}/db" || true
+            ssh -o StrictHostKeyChecking=no -n administrator@${IP} "ipconfig" > "${AGENT_NETWORK_INFO_PATH}/addrs" || true
+        else
+            kubectl exec "${AGENTNAME}" -c antrea-agent -n kube-system -- ovs-ofctl dump-flows br-int > "${AGENT_NETWORK_INFO_PATH}/flows" || true
+            kubectl exec "${AGENTNAME}" -c antrea-agent -n kube-system -- ovs-vsctl show > "${AGENT_NETWORK_INFO_PATH}/db" || true
+            kubectl exec "${AGENTNAME}" -c antrea-agent -n kube-system -- ip a > "${AGENT_NETWORK_INFO_PATH}/addrs" || true
+        fi
+
+        echo "=== Collecting '${AGENTNAME}' antrea agent log after failure ==="
+        ANTREA_AGENT_LOG_PATH="${DEBUG_LOG_PATH}/${AGENTNAME}/antrea_agent_log"
+        mkdir "${ANTREA_AGENT_LOG_PATH}"
+        kubectl logs "${AGENTNAME}" -n kube-system -c antrea-agent > "${ANTREA_AGENT_LOG_PATH}/antrea-agent.log" || true
+        if [[ "${AGENTNAME}" =~ "windows" ]]; then
+            echo "Windows agent doesn't have antrea-ovs container"
+        else
+            kubectl logs "${AGENTNAME}" -n kube-system -c antrea-ovs > "${ANTREA_AGENT_LOG_PATH}/antrea-ovs.log" || true
+        fi
     done
-    tar zcf network_info.tar.gz network_info
 
-    echo "=== Collecting antrea agent log after failure ==="
-    mkdir antrea_agent_log
-    kubectl get pod -n kube-system -l component=antrea-agent --no-headers=true | awk '{print $1}' | while read AGENTNAME; do
-        mkdir antrea_agent_log/$AGENTNAME
-        kubectl logs $AGENTNAME -n kube-system -c antrea-agent > antrea_agent_log/$AGENTNAME/antrea-agent.log || true
-        kubectl logs $AGENTNAME -n kube-system -c antrea-ovs > antrea_agent_log/$AGENTNAME/antrea-ovs.log || true
+    kubectl get nodes -o wide --no-headers=true | awk -v role="$CONTROL_PLANE_NODE_ROLE" '$3 != role && $1 ~ /win/ {print $1}' | while read NODENAME; do
```

Comment:
is `$1 ~ /win/` the best way to check if a Node runs Windows? I guess as long as it works for our testbeds it's fine. An alternative could be to check the `OS-IMAGE` column in the kubectl output.

Reply:
Our Windows VMs in testbed have `win` in their name. But yes using `OS-IMAGE` is more precise.

I will consider using support bundle for this.